### ISSUE TITLE
When origin the inverse side  use for loop collection, the owner's change set is reset.

### ISSUE THREAD

**Issue by kevintoshi0208** [doctrine/orm#7625]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | yes
| Version     | 2.6.3

#### Summary

There is a One To Many RelationShip.
The owing side is Transaction, and the inverse side is Supplier.

When the origin inverse side collection  for loop ,the owning side is reset.

`
     $em = $this->getDoctrine()->getManager();
        /** @var WsTransaction $transaction */

        $transaction = $this->getDoctrine()->getRepository(WsTransaction::class)
            ->find(1);
        $transaction->setSaleAmount(1000);

        $oriSupplier = $transaction->getSupplier();

        //output:  oriSupplier:5
        dump("oriSupplier:".$oriSupplier->getId());

        $supplier = $this->getDoctrine()->getRepository(WsSupplier::class)
            ->find(8);
        $transaction->setSupplier($supplier);

        $em->persist($transaction);

        //output: 8
        dump($transaction->getSupplier()->getId());

        $amount = 0;
        foreach ($oriSupplier->getWsTransactions() as $transaction){
            $amount = $transaction->getSaleAmount();
        }
        $oriSupplier->setOccupiedCredit($amount);

        //expects 8 but output is 5
        dump($transaction->getSupplier()->getId());
        $em->persist($transaction);

        $em->flush();

        //expects 8  but output is 5
        dump($transaction->getSupplier()->getId());
`
How can I do that ?

**Comment by Ocramius** [comment]
Could you reduce the code example a bit? I'm not sure if I understand the issue. Is the association being overwritten by lazy loading?

**Comment by kevintoshi0208** [comment]
Yes,The association is overwritten when I get the association at the inverse side ,even though I set the fetch option to EAGER.

**Comment by kevintoshi0208** [comment]
Thank you for your keyword. I solved it  by setting the fetch options of both side to EAGER .
